runners/spark/src/main/java/org/apache/beam/runners/spark/io/SourceDStream.java

```diff
@@ -84,6 +85,10 @@
     SparkPipelineOptions options = runtimeContext.getPipelineOptions().as(
         SparkPipelineOptions.class);
 
+    // Reader cache interval is set to expire readers if they have not been accessed in the last
+    // microbatch. 50% of batch interval is added to accommodate latency.
```

Comment:
It appears as if there's no reason to invalidate readers at all (definitely not after half a batch..),  but we know that's not the case, right ? we're considering failures, re-execution on same JVM, different JVM, etc. 
Maybe worth relating to that in some way in the comments.